### PR TITLE
Fix windowsize for fullscreen windows on windowcreation on Windows

### LINKED ISSUE #72596: "phantom viewport" when launching maximized interferes with mouse input
### Godot version

4.0.beta17

### System information

Windows 10/11

### Issue description

When project settings are set to "Maximized", "Fullscreen", or "Exclusive Fullscreen", and the base viewport size is smaller than the resulting window size, the following occurs:
Moving the mouse into the top left of the window (into a region the size of the base viewport size) causes mouse enter/exit behavior (including both Controls and Node2D object picking) to stop working outside of that region. Clicking on Controls still works. Resizing the window fixes the error until the project is closed.

### Steps to reproduce

- Make new project
- Set base viewport size in Project Settings to something smaller than your monitor resolution
- Set the window's launch mode to Maximized, Fullscreen, or Exclusive Fullscreen
- Add a few Controls to the scene
- Run the project
- Hover over the top left portion of the screen
- Hover outside of the top left portion of the screen

### Minimal reproduction project

[Test.zip](https://github.com/godotengine/godot/files/10569454/Test.zip)

### PR DESCRIPTION
**Pull request by Sauermann**
Fixes #72596

Thanks to the writer of the FIXME comment. It helped a lot!

### CODE REVIEW

File: platform/windows/display_server_windows.cpp [hunk 1]
```diff
@@ -3959,10 +3959,18 @@ DisplayServer::WindowID DisplayServerWindows::_create_window(WindowMode p_mode,
...
-		wd.last_pos = p_rect.position;
-		wd.width = p_rect.size.width;
-		wd.height = p_rect.size.height;
+		if (p_mode == WINDOW_MODE_FULLSCREEN || p_mode == WINDOW_MODE_EXCLUSIVE_FULLSCREEN || p_mode == WINDOW_MODE_MAXIMIZED) {
+			RECT r;
+			if (GetWindowRect(wd.hWnd, &r)) {
+				wd.last_pos = Point2(r.left, r.top);
+				wd.width = r.right - r.left;
+				wd.height = r.bottom - r.top;
+			}
```
Comment (bruvzg) [inline platform/windows/display_server_windows.cpp:3968]: This should be client rect, with the position adjusted to account for the different screen coordinate system used by Godot and Windows.

```suggestion
			GetClientRect(wd.hWnd, &r);
			ClientToScreen(wd.hWnd, (POINT *)&r.left);
			ClientToScreen(wd.hWnd, (POINT *)&r.right);
			wd.last_pos = Point2i(rect.left, rect.top) - _get_screens_origin();
			wd.width = r.right;
			wd.height = r.bottom;
```

Comment (Sauermann) [reply platform/windows/display_server_windows.cpp]: Thank you for the correction. I just replaced `rect.left` by `r.left`.

File: platform/windows/display_server_windows.cpp [hunk 2]
```diff
@@ -3959,10 +3959,19 @@ DisplayServer::WindowID DisplayServerWindows::_create_window(WindowMode p_mode,
...
-		wd.width = p_rect.size.width;
-		wd.height = p_rect.size.height;
+		if (p_mode == WINDOW_MODE_FULLSCREEN || p_mode == WINDOW_MODE_EXCLUSIVE_FULLSCREEN || p_mode == WINDOW_MODE_MAXIMIZED) {
+			RECT r;
+			GetClientRect(wd.hWnd, &r);
+			ClientToScreen(wd.hWnd, (POINT *)&r.left);
+			ClientToScreen(wd.hWnd, (POINT *)&r.right);
+			wd.last_pos = Point2i(r.left, r.top) - _get_screens_origin();
+			wd.width = r.right;
+			wd.height = r.bottom;
```
Comment (bruvzg) [inline platform/windows/display_server_windows.cpp:3969]:
```suggestion
			wd.width = r.right - r.left;
			wd.height = r.bottom - r.top;
```
These two lines actually were correct, usually client rect starts at (0, 0), but it was changed by `ClientToScreen`.

**Review by bruvzg — APPROVED** [review]

**Review by bruvzg — APPROVED** [review]

File: platform/windows/display_server_windows.cpp (hunk 2 above)
Comment (Sauermann) [reply platform/windows/display_server_windows.cpp]: I have adjusted them. Thanks!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**